accepted/00052-new-package-predownloading.md

```diff
@@ -0,0 +1,85 @@
+  - package installation: https://server.tf.local/rhn/systems/details/packages/InstallPackages.do?sid=1000010000&
+  - package upgrade: https://server.tf.local/rhn/systems/details/packages/UpgradableList.do?sid=1000010000&
+
+- SSM:
+  - patch application: https://server.tf.local/rhn/systems/ssm/ListErrata.do
+  - package installation: https://server.tf.local/rhn/ssm/PackageInstall.do
+  - package upgrade: https://server.tf.local/rhn/ssm/PackageUpgrade.do
+
+*Links above assume `server.tf.local` as the server location and `1000010000` as a minion ID.*
+
+### API
+
+The following calls will need a new optional `downloadOnly` boolean flag:
+- `actionchain.addErrataUpdate`
+- `actionchain.addPackageInstall`
+- `actionchain.addPackageUpgrade`
+- `actionchain.schedulePackageInstall`
+- `actionchain.schedulePackageInstallByNevra`
+- `system.schedulePackageInstall`
+- `system.schedulePackageInstallByNevra`
+- `system.scheduleApplyErrata`
+- `systemgroup.scheduleApplyErrataToActive`
+
+## Implementation
+- new checkbox item will be added to current JSPs/React components
+  - checkbox has to be shown only if applicable systems are all minions
+- existing Java handling code will receive the new flag, pass it down to Salt calling code in order to apply
+  - [pkgdownload.sls](https://github.com/uyuni-project/uyuni/blob/c8ffe6b9425392f5235864ad070646bb8ebc2ecb/susemanager-utils/susemanager-sls/salt/packages/pkgdownload.sls) instead of [pkginstall.sls](https://github.com/uyuni-project/uyuni/blob/c8ffe6b9425392f5235864ad070646bb8ebc2ecb/susemanager-utils/susemanager-sls/salt/packages/pkginstall.sls) if set
+  - [patchdownload.sls](https://github.com/uyuni-project/uyuni/blob/c8ffe6b9425392f5235864ad070646bb8ebc2ecb/susemanager-utils/susemanager-sls/salt/packages/patchdownload.sls) instead of [patchinstall.sls](https://github.com/uyuni-project/uyuni/blob/c8ffe6b9425392f5235864ad070646bb8ebc2ecb/susemanager-utils/susemanager-sls/salt/packages/patchinstall.sls) if set
+  - alternatively, those files might be unified and use `pkg.downloaded` instead of `pkg.installed` conditionally (resp. `pkg.patch_downloaded` instead of `pkg.patch_installed`)
+- existing configuration options are to be deprecated (in SUSE Manager supported versions) or removed
+  - existing code handling the configuration options can be removed from Uyuni
+
+# Drawbacks
+ - this is limited to Salt minions. Traditional clients retain their existing mechanism and are not affected
+ - as the `pkg.downloaded` state [is not yet supported in Salt for Debian systems](https://docs.saltstack.com/en/2018.3/ref/states/all/salt.states.pkg.html#salt.states.pkg.downloaded), this remains an exclusive for `yum` and `zypper` (this RFC does not change this aspect)
+ - current users of the existing feature will have to adapt their workflows
+
+# Alternatives
+- add UI pages and API calls to define "download time windows". Allow users to Actions to "download time windows" at scheduling time
```

Comment:
Why not uniform the name? "Content staging for minions"

Reply:
Fixed.